python/semantic_kernel/utils/validation.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from re import match as re_match
+from typing import Optional
+
+
+def validate_skill_name(value: Optional[str]) -> None:
+    """
+    Validates that the skill name is valid.
+
+    Valid skill names are non-empty and
+    match the regex: [0-9A-Za-z_]*
+
+    :param value: The skill name to validate.
+
+    :raises ValueError: If the skill name is invalid.
+    """
+    if not value:
+        raise ValueError("The skill name cannot be `None` or empty")
+
+    SKILL_NAME_REGEX = r"^[0-9A-Za-z_]*$"
+    if not re_match(SKILL_NAME_REGEX, value):
+        raise ValueError(
+            f"Invalid skill name: {value}. Skill "
+            f"names must match the regex: {SKILL_NAME_REGEX}"
```

Comment:
This is a bit weak as a user-facing error; why not explain that skill names may only contain ASCII letters, digits, and underscores?

Reply:
Agreed, edited to match your suggestion.